src/gui/editorwidgets/core/qgseditorwidgetfactory.cpp

```diff
@@ -123,3 +123,27 @@ unsigned int QgsEditorWidgetFactory::fieldScore( const QgsVectorLayer* vl, int f
   return 5;
 }
 
+void QgsEditorWidgetFactory::config2xml( const QgsEditorWidgetConfig& config, QDomElement& configElement, const QString& fieldName )
+{
+  const QVariant value = config.value( fieldName );
+  if ( value.isValid() )
+  {
+    if ( value.type() == QVariant::Bool )
+    {
+      configElement.setAttribute( fieldName, value.toBool() ? "1" : "0" );
+    }
+    else
+    {
+      configElement.setAttribute( fieldName, value.toString() );
+    }
+  }
+}
+
+void QgsEditorWidgetFactory::xml2config( const QDomElement& configElement, QgsEditorWidgetConfig& config, const QString& fieldName )
+{
+  const QString value = configElement.attribute( fieldName );
+  if ( !value.isNull() )
+  {
+    config.insert( fieldName, value );
+  }
```

Comment:
I think the application had knowledge of type information before, see e.g. https://github.com/qgis/QGIS/pull/3585/files#diff-22d1ee53da45d48890d2e7e5d0545e95L61 

I'm not sure if something relied on it further down the road but it's worth taking into account